pkg/rootfs/apply.go

```diff
@@ -121,6 +121,9 @@ func applyLayers(ctx context.Context, layers []Layer, chain []digest.Digest, sn
 		err     error
 	)
 
+	labels := make(map[string]string)
+	labels["containerd.io/snapshot.ref"] = chainID.String()
```

Comment:
this label is added here https://github.com/containerd/containerd/blob/main/core/unpack/unpacker.go#L311 already.

Is the label only missing when you pull with ctr (not crictl for example) ? It's probably because ctr uses a different unpack method from here https://github.com/containerd/containerd/blob/main/client/image.go#L290, which doesn't add the snapshot.ref label.

Does it make more sense to add the labels at https://github.com/containerd/containerd/blob/main/client/image.go#L336 via SnapshotOpts, so that it's only triggered by ctr code path?

Reply:
> 
> Is the label only missing when you pull with ctr (not crictl for example) ? It's probably because ctr uses a different unpack method from here https://github.com/containerd/containerd/blob/main/client/image.go#L290, which doesn't add the snapshot.ref label.

Yes, I think only `ctr` is impacted. Honestly, I'm still learning containerd code in my limited spare time mostly due to `erofs snapshotter` stuff.

So sorry for the limited knowledge.

> 
> Does it make more sense to add the labels at https://github.com/containerd/containerd/blob/main/client/image.go#L336 via SnapshotOpts, so that it's only triggered by ctr code path?

Okay, I will try to add there.  Thanks for your suggestion!

Reply:
Hi @ruiwen-zhao , could you take another look? 😄